src/main/java/com/iheartradio/m3u8/MediaPlaylistTagHandler.java

```diff
@@ -24,9 +26,109 @@ private void validateNotMaster(ParseState state) throws ParseException {
             throw ParseException.create(ParseExceptionType.MEDIA_IN_MASTER, getTag());
         }
     }
-
+    
     // media playlist tags
+    
+    static final IExtTagHandler EXT_X_ENDLIST = new MediaPlaylistTagHandler() {
+        @Override
+        public String getTag() {
+            return Constants.EXT_X_ENDLIST_TAG;
+        }
+
+        @Override
+        boolean hasData() {
+            return false;
+        }
+
+        @Override
+        public void handle(String line, ParseState state) throws ParseException {
+            super.handle(line, state);
+
+            match(Constants.EXT_X_ENDLIST_PATTERN, line);
+            //TODO we should ensure that no new items are added beyond this point to the playlist
+        }
+    };
+    
+    static final IExtTagHandler EXT_X_I_FRAMES_ONLY = new MediaPlaylistTagHandler() {
+        @Override
+        public String getTag() {
+            return Constants.EXT_X_I_FRAMES_ONLY_TAG;
+        }
+
+        @Override
+        boolean hasData() {
+            return false;
+        }
+
+        @Override
+        public void handle(String line, ParseState state) throws ParseException {
+            super.handle(line, state);
+            
+            final Matcher matcher = match(Constants.EXT_X_I_FRAMES_ONLY_PATTERN, line);
+            
+            if (state.getCompatibilityVersion() < 4) {
+                throw ParseException.create(ParseExceptionType.REQUIRES_PROTOCOL_VERSION_4_OR_HIGHER, getTag());
+            }
+        }
+    };
+    
+    static final IExtTagHandler EXT_X_PLAYLIST_TYPE = new MediaPlaylistTagHandler() {
+        @Override
+        public String getTag() {
+            return Constants.EXT_X_PLAYLIST_TYPE_TAG;
+        }
+
+        @Override
+        boolean hasData() {
+            return true;
+        }
+
+        @Override
+        public void handle(String line, ParseState state) throws ParseException {
+            super.handle(line, state);
+
+            final Matcher matcher = match(Constants.EXT_X_PLAYLIST_TYPE_PATTERN, line);
 
+            if (state.getMedia().targetDuration != null) {
+                throw ParseException.create(ParseExceptionType.MULTIPLE_EXT_TAG_INSTANCES, getTag(), line);
+            }
+
+            state.getMedia().playlistType = ParseUtil.parseEnum(matcher.group(1), PlaylistType.class, getTag());
+        }
+    };
+    
+    static final IExtTagHandler EXT_X_START = new MediaPlaylistTagHandler() {
+        @Override
+        public String getTag() {
+            return Constants.EXT_X_START_TAG;
+        }
+
+        @Override
+        boolean hasData() {
+            return true;
+        }
+
+        @Override
+        public void handle(String line, ParseState state) throws ParseException {
+            super.handle(line, state);
+
+            final Matcher matcher = match(Constants.EXT_X_START_PATTERN, line);
```

Comment:
EXT-X-START is specified as having an attribute list. Can you implement this using `parseAttributes` instead of `match`?